### PR TITLE
 ncm-metaconfig: haproxy - support several options like ssl, sticky tables, peers, ...

### PR DESCRIPTION
**Pull request by wdpypere**
support default ssl configuration options

### CODE REVIEW

File: ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan [hunk 1]
```diff
@@ -21,6 +21,11 @@ type haproxy_service_global_config = {
     'daemon' : string = ''
     'pidfile' : string = '/var/run/haproxy.pid'
     'chroot' : string = '/var/lib/haproxy'
+
+    'ssl-default-bind-ciphers' ? string
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan]: Could this at least be `string_non_whitespace`?

**Comment by wdpypere** [comment]
@jrha remark addressed.

File: ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan [hunk 2]
```diff
@@ -21,6 +21,11 @@ type haproxy_service_global_config = {
     'daemon' : string = ''
     'pidfile' : string = '/var/run/haproxy.pid'
     'chroot' : string = '/var/lib/haproxy'
+
+    'ssl-default-bind-ciphers' ? string_non_whitespace
```
Comment (kwaegema) [inline ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan]: shouldnt this be an array?

File: ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan [hunk 3]
```diff
@@ -21,6 +21,11 @@ type haproxy_service_global_config = {
...
+    'ssl-default-bind-options' ? string_non_whitespace
```
Comment (kwaegema) [inline ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan]: same for the others

File: ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan (hunk 2 above)
Comment (wdpypere) [reply ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan]: @kwaegema it looks like `   ssl-default-server-ciphers ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE...`
So an array will not do afaik.

Comment (jrha) [reply ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan]: Or in other words: it should be, but the TT file will need updating.

Comment (wdpypere) [reply ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan]: yes. I will make an update to this PR to fix all this.

Comment (jrha) [reply ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan]: Awesome thanks!

**Comment by wdpypere** [comment]
@jrha I updated the PR with decent TT implementation for the cipher stuff. In the mean time I added some other options we need as well.
Travis still does not like `-` in comments.

File: ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan [hunk 4]
```diff
@@ -21,6 +21,7 @@ type haproxy_service_global_config = {
     'daemon' : string = ''
     'pidfile' : string = '/var/run/haproxy.pid'
     'chroot' : string = '/var/lib/haproxy'
+    'log-send-hostname' ? string
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan]: `type_hostname`?

Comment (wdpypere) [reply ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan]: actually, what I'm trying to achieve is either:
```
log-send-hostname somestringyoucaninvent
```
and
```
log-send-hostname
```
the latter is directing haproxy to guess the hostname itself.

Comment (jrha) [reply ncm-metaconfig/src/main/metaconfig/haproxy/pan/schema.pan]: Hmm, ok!

**Comment by kwaegema** [comment]
I know you rather have nothing to do with it, but please add the config you add in tt file to the tests.
you only need to add the config valies to  ncm-metaconfig/src/main/metaconfig/haproxy/tests/profiles/config.pan (see eg https://github.com/quattor/configuration-modules-core/pull/1267/files) and then run the perl oneliner on the output of mvn_test that generates the regexfiles. From my history: `mvn_test service-haproxy 2>&1 >/dev/null  | perl -e 'while (<>) { s/^# //; s/([(){}|[\]+\^\*\$\?#])/\\\1/g; s/^/^/; s/$/\$/; s/ {16}/\\s{16}/;s/ {12}/\\s{12}/; s/ {8}/\\s{8}/; s/ {4}/\\s{4}/; print; }'`

**Comment by jrha** [comment]
Travis errors are safe to ignore

**Pull request merged**